spec/default_formatter_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "helper"
+require "simplecov_json_formatter"
+
+describe SimpleCov::Formatter do
+  describe ".from_env" do
+    context "when CC_TEST_REPORTER_ID environment variable is set" do
+      before do
+        ENV["CC_TEST_REPORTER_ID"] = "4c9f1de6193f30799e9a5d5c082692abecc1fd2c6aa62c621af7b2a910761970"
+      end
+
+      it "returns a MultiFormatter instance with the HTML and JSON formatter" do
+        expect(SimpleCov::Formatter::MultiFormatter).to receive(:new).with([
```

Comment:
I think you could make this a lot easier to test by:

1. passing in the ENV as an argument, that way it's easy to test and we don't need to mess with global state that might leak into other tests for weird results (if the tests here are run in random order so that the nil one isn't run some test further down the line might still have it set)
2. just returning the array of formatters to use instead of the Multiformatter and doing that on the call site, that way you avoid the stubbing :)

Reply:
Thanks, both suggestions have simplified testing a lot !